frame/support/src/traits/hooks.rs

```diff
@@ -336,42 +334,41 @@ pub trait Hooks<BlockNumber> {
-	/// of those long-running computations back on chain.
+	/// Implementing this function on a pallet allows you to perform long-running tasks that are
+	/// dispatched as separate threads, and entirely independent of the main wasm runtime.
 	///
-	/// NOTE: This function runs off-chain, so it can access the block state,
-	/// but cannot preform any alterations. More specifically alterations are
-	/// not forbidden, but they are not persisted in any way after the worker
-	/// has finished.
+	/// This function can freely read from the state, but any change it makes to the state is
+	/// meaningless. Writes can be pushed back to the chain by submitting extrinsics from the
+	/// offchain worker to the transaction pool. See `pallet-example-offchain-worker` for more
+	/// details on this.
 	///
-	/// This function is being called after every block import (when fully synced).
+	/// Moreover, the code in this function has access to a wider range of host functions in
+	/// [`sp-io`], namely [`sp_io::offchain`]. This includes exotic operations such as HTTP calls
+	/// that are not really possible in the rest of the runtime code.
 	///
-	/// Implement this and use any of the `Offchain` `sp_io` set of APIs
-	/// to perform off-chain computations, calls and submit transactions
-	/// with results to trigger any on-chain changes.
-	/// Any state alterations are lost and are not persisted.
+	/// The execution of this hook is entirely optional and is left at the discretion of the node
+	/// side software, and its configuration. In a normal substrate-cli, look for the CLI flags
+	/// related to offchain-workers to learn more.
 	fn offchain_worker(_n: BlockNumber) {}
 
-	/// Run integrity test.
+	/// Check the integrity of this pallet's configuration.
+	///
+	/// Any code located in this hook is placed in an auto-generated test, and generated as a part
+	/// of [`construct_runtime`]'s expansion. Look for a test case with a name along the lines of:
+	/// `__construct_runtime_integrity_test`.
 	///
-	/// The test is not executed in a externalities provided environment.
+	/// This hook is suitable location to ensure value values/types provided to the `Config` trait
+	/// of the pallet can be tested for correctness. For example, if two `type Foo: Get<u32>` and
+	/// `type Bar: Get<u32>` where `Foo::get()` must always be greater than `Bar::get()`, such
+	/// checks can be asserted upon here.
+	///
+	/// Note that this hook is not executed in an externality environment, so if access to state is
+	/// needed, the code should be wrapped in `sp_io::TestExternalities`.
 	fn integrity_test() {}
 }
 
-/// A trait to define the build function of a genesis config for both runtime and pallets.
-///
-/// Replaces deprecated [`GenesisBuild<T,I>`].
-pub trait BuildGenesisConfig: Default + sp_runtime::traits::MaybeSerializeDeserialize {
-	/// The build function puts initial `GenesisConfig` keys/values pairs into the storage.
-	fn build(&self);
-}
```

Comment:
is it removed by mistake ?

Reply:
No I must have been out of sync with master 🙈

Reply:
will fix soon.